### ISSUE TITLE
Reading logs from multiple files inside the same container.

### ISSUE THREAD

**Issue by saif-cl0ud** [splunk/splunk-connect-for-kubernetes#246]
I want to read and export logs from multiple logs file reside inside the same container. But the logging agent is only scraping logs at the node level and defaults logs inside the containers folder.
Their is anyway to read logs directly inside the container and export it to Splunk.

**Comment by matthewmodestino** [comment]
Yes, you can mount a volume into the collector ad read from it, or you can deploy a sidecar with the fluentd-hec image or the splunk universal forwarder docker image.

**Comment by saif-cl0ud** [comment]
Do we have a option to configure sidecar container from splunk connect for kubernetes or we need to do it manually? Any reference for manual process?


On 24-Sep-2019, at 6:58 AM, Matthew Modestino <notifications@github.com<mailto:notifications@github.com>> wrote:


Yes, you can mount a volume into the collector ad read from it, or you can deploy a sidecar with the fluentd-hec image or the splunk universal forwarder docker image.

**Comment by matthewmodestino** [comment]
Splunk Connect for Kubernetes would be the way to go to avoid the sidecar, by collecting data from the volume you would mount into the logging pod.

If you want to deploy a sidecar, i would recommend using out fluentd-hec image or the docker-splunk uf. Will share example.

**Comment by saif-cl0ud** [comment]
By mounting volume , you meant host(worker nodes) volume mounted on container from where logging agent will read log file. The file path will be provided in values.yaml for logging chart.
Or any other persistent volume mounted on both my application container and logging agent pod  from where the agent will read the log.

**Comment by saif-cl0ud** [comment]
Another question is if we are mounting host volume for different log files of my application container. This application container has replica count 3 and 2 of them is in same worker nodes ie same host volume. Then they will be using same log files to write log and may create issue like file lock.

**Comment by deepak4babu** [comment]
@matthewmodestino Hi Was any example shared to deploy fluentd/ splunk-hf as side car

**Comment by Ian-An** [comment]
I notice the default is mounting /var/log to the collector, and I wonder if I can mount it another direcotory /var/xx/logs? if so, could you @matthewmodestino provide an example of it?

**Comment by rockb1017** [comment]
@saif-cl0ud 
this issue has been a while. 
are you still having a trouble? 

@deepak4babu 
deploying agent as a sidecar is complicated and it is not shared in this project.  

@Ian-An 
you can just follow the configurations for `/var/log` path for any other directories. Please see fluentd docs and also consult your k8s admin.

**Comment by saif-cl0ud** [comment]
We have to deploy a sidecar fluentd container to read multiple log files. Then sent the log details to splunk.
